server/service/src/document/patient/query.rs

```diff
@@ -85,6 +88,35 @@ pub fn get_patients(
         .collect())
 }
 
+pub fn get_patient_programs(
+    ctx: &ServiceContext,
+    store_id: &str,
+    patient_id: &str,
+) -> Result<Vec<Document>, RepositoryError> {
+    DocumentRepository::new(&ctx.connection).query(Some(
+        DocumentFilter::new()
+            .name(StringFilter::starts_with(&patient_program_doc_name(
+                patient_id, "",
+            )))
+            .store_id(EqualFilter::equal_to(store_id)),
+    ))
+}
+
+pub fn get_patient_program_encounters(
+    ctx: &ServiceContext,
+    store_id: &str,
+    patient_id: &str,
+    program: &str,
+) -> Result<Vec<Document>, RepositoryError> {
+    DocumentRepository::new(&ctx.connection).query(Some(
+        DocumentFilter::new()
+            .name(StringFilter::starts_with(
+                &patient_program_encounter_doc_name(patient_id, program, ""),
```

Comment:
I still think this is anti pattern for relational databases, and requires inspecting code to figure out how data is stored. I would like to revisit this sometime (and the requirements for it).

Reply:
You mean patient_program_encounter_doc_name? we need a deterministic way to find a document name, e.g. Creating two HIVPrograms on two sites at the same sites should result in a single document